bravado/http_future.py

```diff
@@ -141,25 +144,25 @@ def response(self, timeout=None, fallback_result=None, exceptions_to_catch=FALLB
         except exceptions_to_catch as e:
             if end_time is None:
                 end_time = monotonic.monotonic()
-            exc_info = sys.exc_info()
+            caught_exception = True
             if (
                 fallback_result and self.operation
                 and not self.operation.swagger_spec.config['bravado'].disable_fallback_results
             ):
                 swagger_result = fallback_result(e)
             else:
-                six.reraise(*exc_info)
+                six.reraise(*sys.exc_info())
 
         metadata_class = self.operation.swagger_spec.config['bravado'].response_metadata_class
         response_metadata = metadata_class(
             incoming_response=incoming_response,
             swagger_result=swagger_result,
             elapsed_time=end_time - self._start_time,
-            exc_info=exc_info,
+            handled_exception_info=sys.exc_info() if caught_exception else None,
```

Comment:
Is that going to work? When I try `sys.exc_info` outside of the `except` block I can't get to it anymore (trying with python 2.7.6):

```
>>> try:
...     raise ValueError('hi')
... except:
...     pass
...
>>> import sys; print(sys.exc_info())
(None, None, None)
>>> try:
...     raise ValueError('hi')
... except:
...     import sys; print(sys.exc_info())
...
(<type 'exceptions.ValueError'>, ValueError('hi',), <traceback object at 0x7f84248d23f8>)
```

Reply:
Yeah, looks like it doesn't work. :( Interestingly, on Travis the test only fails on Python 3... not sure why.
I've rewritten it so that we'll have a string representation of the traceback, as we only need it for logging anyway.